imagetest/test_suites/README.md

```diff
@@ -32,7 +32,54 @@ new space.
 disk and reboot the VM via the API. Wait for the VM to boot again, and validate
 the new size as reported by the operating system matches the expected size.
 
-### Test suite: image\_boot
+## Test suite: hostnamevalidation ##
+
+Tests which verify that the metadata hostname is created and works with the DNS record.
+
+#### TestHostname
+Test that the system hostname is correctly set.
+
+- <b>Background</b>: The hostname is one of many pieces of 'dynamic' configuration that supported
+GCE Images will set for you. This is compared to the
+'static' configuration which is present on the image to be tested. Dynamic
+configuration allows a single GCE Image to be used on many VMs without
+pre-modification.
+
+- <b>Test logic</b>: Retrieve the intended FQDN from metadata (which is authoritative) and
+compare the hostname part of it (first label) to the currently set hostname as
+returned by the kernel.
+
+#### TestFQDN
+Test that the fully-qualified domain name is correctly set.
+
+- <b>Background</b>: The FQDN is a complicated concept in Linux operating systems, and setting it in
+an incorrect way can lead to unexpected behavior in some software.
+
+- <b>Test logic</b>: Retrieve the intended FQDN from metadata and compare the full value to the
+output of `/bin/hostname -A`. See `man 1 hostname` for more details.
+
+#### TestCustomHostname
+Test that custom hostnames are correctly set.
+
+- <b>Background</b>: The hostname for a VM matches the name of the VM by default. However, if you
+specify a custom hostname at instance creation time, this will be used instead.
```

Comment:
The test is called customhostname but what we are actually testing should be referred to as a custom domain name, not a custom hostname.